File: api/csharp/ToolGood.TextFilter.Api/Datas/Requests/TextFilterRequest.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ToolGood.TextFilter.Api.Datas.Requests
{
    public class TextFilterRequest
    {
        /// <summary>
        /// 需要检测的文本
        /// </summary>
        public string Txt { get; set; }

        /// <summary>
        /// 是否跳过 Bidi 字符，默认 false
        /// </summary>
        public bool SkipBidi { get; set; }
        /// <summary>
        /// 只显示位置，不显示匹配文字
        /// </summary>
        public bool OnlyPosition { get; set; }


        public TextFilterRequest()
        {
            SkipBidi = false;
        }
        /// <summary>
        /// 
        /// </summary>
        /// <param name="txt">需要检测的文本</param>
        /// <param name="skipBidi">是否跳过 Bidi 字符</param>
        /// <param name="onlyPosition">是否只显示位置，不显示匹配文字</param>
        public TextFilterRequest(string txt,bool skipBidi = false, bool onlyPosition = false)
        {
            Txt = txt;
            SkipBidi = skipBidi;
            OnlyPosition = onlyPosition;

        }

#if GRPC
        public TextFindAllGrpcRequest ToGrpcRequest()
        {
            return new TextFindAllGrpcRequest() {
                Txt = this.Txt,
                SkipBidi = this.SkipBidi,
                OnlyPosition = OnlyPosition,

            };
        }

#endif


    }


}

File: api/java/toolgood.textfilter.api/src/main/java/toolgood/textfilter/api/Datas/Requests/TextFilterRequest.java
package toolgood.textfilter.api.Datas.Requests;

import toolgood.textfilter.api.GrpcBase.TextFilter.TextFindAllGrpcRequest;

public class TextFilterRequest {
    /**
     * 需要检测的文本
     */
    public String Txt;

    /**
     * 是否跳过 Bidi 字符，默认 false
     */
    public boolean SkipBidi;
    /**
     * 只显示位置，不显示匹配文字
     */
    public boolean OnlyPosition;

    public TextFilterRequest() {
        SkipBidi = false;
    }

    /**
     * 
     * @param txt          需要检测的文本
     * @param skipBidi     是否跳过 Bidi 字符
     * @param onlyPosition 是否只显示位置，不显示匹配文字
     */
    public TextFilterRequest(String txt, boolean skipBidi, boolean onlyPosition) {
        Txt = txt;
        SkipBidi = skipBidi;
        OnlyPosition = onlyPosition;
    }

    public TextFindAllGrpcRequest ToGrpcRequest() {
        return TextFindAllGrpcRequest.newBuilder().setTxt(Txt).setSkipBidi(SkipBidi).setOnlyPosition(OnlyPosition)
                .build();
    }

}
